pkg/action/executor/resize_container_util.go

```diff
@@ -168,81 +147,118 @@ func genMemoryQuantity(newValue float64) (resource.Quantity, error) {
+func resizeSingleContainer(client *kclient.Clientset, inputPod *k8sapi.Pod, spec *containerResizeSpec) (*k8sapi.Pod, error) {
+	// check parent controller of the input pod
+	fullName := util.BuildIdentifier(inputPod.Namespace, inputPod.Name)
+	parentKind, parentName, err := podutil.GetPodParentInfo(inputPod)
+	if err != nil {
+		glog.Errorf("Resize action failed: failed to get pod[%s] parent info: %v.", fullName, err)
+		return nil, err
+	}
+	if !util.SupportedParent(parentKind) {
+		err = fmt.Errorf("parent kind %v is not supported", parentKind)
+		glog.Errorf("Resize action aborted: %v.", err)
+		return nil, err
+	}
+
+	id := fmt.Sprintf("%s/%s-%d", inputPod.Namespace, inputPod.Name, spec.Index)
+	if parentKind == "" {
+		glog.V(2).Infof("Begin to resize bare pod container[%s].", id)
+	} else {
+		glog.V(2).Infof("Begin to resize container[%s] parent=%s/%s.",
+			id, parentKind, parentName)
+	}
 
-	podClient := client.CoreV1().Pods(tpod.Namespace)
+	// get the current Pod
+	podClient := client.CoreV1().Pods(inputPod.Namespace)
 	if podClient == nil {
-		err := fmt.Errorf("Cannot get Pod client for nameSpace: %v", tpod.Namespace)
-		glog.Error(err)
+		err := fmt.Errorf("failed to get pod client for namespace: %v", inputPod.Namespace)
+		glog.Errorf("Failed to resize container %s: %v", id, err)
 		return nil, err
 	}
-
-	// 0. get the latest Pod
-	var err error
-	var pod *k8sapi.Pod
-	if pod, err = podClient.Get(tpod.Name, metav1.GetOptions{}); err != nil {
-		glog.Errorf("Failed to get latest pod %v: %v", id, err)
+	originalPod, err := podClient.Get(inputPod.Name, metav1.GetOptions{})
+	if err != nil {
+		glog.Errorf("Failed to resize container %s: %v", id, err)
 		return nil, err
 	}
-	labels := pod.Labels
+	labels := originalPod.Labels
 
-	// 1. create a clone pod
-	npod, changed, err := createResizePod(client, pod, spec)
+	// create a clone pod with new size
+	clonePod, changed, err := clonePodWithNewSize(client, originalPod, spec)
 	if err != nil {
-		glog.Errorf("resizeContainer failed[%s]: failed to create a resized pod: %v", id, err)
+		glog.Errorf("Failed to clone pod %s with new size: %v", id, err)
 		return nil, err
 	}
 
 	if !changed {
-		glog.Warningf("resizeContainer aborted[%s]: no need do resize container.", id)
-		return nil, fmt.Errorf("Resize aborted due to not enough change")
+		glog.Warningf("No need to resize container %s. Not enough change.", id)
+		return nil, fmt.Errorf("resize aborted due to not enough change")
 	}
-	//delete the clone pod if this action fails
+
+	// delete the clone pod if this action fails
 	flag := false
 	defer func() {
 		if !flag {
-			glog.Errorf("resizeContainer failed, begin to delete cloned pod: %v/%v.", npod.Namespace, npod.Name)
-			delOpt := &metav1.DeleteOptions{}
-			podClient.Delete(npod.Name, delOpt)
+			glog.Errorf("resizeContainer failed, begin to delete cloned pod: %v/%v.", clonePod.Namespace, clonePod.Name)
+			_ = podClient.Delete(clonePod.Name, &metav1.DeleteOptions{})
```

Comment:
The `podClient.Delete()` would return an error right?  So we're not checking?  If not, then why doing the assignment?

Reply:
Originally I did the assignment to get rid of the warnings in IDE. But you are right, we should do the error checking here.  This is the place where we delete the cloned pod when resizing a single container fails. If there is error when deleting the cloned pod, there is nothing we can do, but we should still log a warning message. Updated the code.